### ISSUE TITLE
How to ensure that training data is shuffled once within each epoch?

### ISSUE THREAD

**Issue by petrama** [balancap/SSD-Tensorflow#20]
It would be cool to have support for shuffling training data within each epoch. I mean on a behavior similar to:

string_input_producer has options for shuffling and setting a maximum number of epochs. A queue runner adds the whole list of filenames to the queue once for each epoch, shuffling the filenames within an epoch if shuffle=True. This procedure provides a uniform sampling of files, so that examples are not under- or over- sampled relative to each other. 

It seems that this way every batch is sampled completely random from the set. Lets say I have images 1 2 3 4 in the training set and batch_size=2. With this code, when training, I can get batches:

1 2| 1 3| 1 4| 2 3

so in two epochs, net has seen example 1 three times, and 4 only one time.
